From 71c5ec5f9f65e75ce1105e78b1d1ff41962fc32e Mon Sep 17 00:00:00 2001
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Mon, 26 Oct 2015 17:30:01 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Do not overwrite primary plane
 register

Plane configurations like Gamma are not sent in each flip request. Keep
older reg.cntr value of primary plane instead of overwriting it.

SQUASHME! - This patch should be squashed to the following patch:
	commit: 213ffb1b1918655f62cf3645357300e0e6da86ad
	Author: Gajanan Bhat <gajanan.bhat@intel.com>
	Date:   Fri Oct 9 12:01:43 2015 +0530
	REVERTME [VPG]: drm/i915: Re-enable primary if flip done is set on primary plane

Change-Id: I4ffee669532d1d1a8267a48ebe3bac61415cae11
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6830
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index e3d5c5c..0c68911 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -10791,7 +10791,7 @@ static int intel_crtc_page_flip(struct drm_crtc *crtc,
 	/* If page flip event flag is set, primary should be enabled */
 	if (page_flip_flags & DRM_MODE_PAGE_FLIP_EVENT) {
 		intel_crtc->pri_update = true;
-		intel_crtc->reg.cntr = DISPLAY_PLANE_ENABLE;
+		intel_crtc->reg.cntr |= DISPLAY_PLANE_ENABLE;
 		dev_priv->pipe_plane_stat |=
 			VLV_UPDATEPLANE_STAT_PRIM_PER_PIPE(intel_crtc->pipe);
 	}
-- 
1.9.1

